### PR TITLE
Adds the start of a new sample showcasing Cupertino widgets.

### PR DESCRIPTION
**Pull request by RedBrogdon**

### CODE REVIEW

File: veggieseasons/lib/data/veggie.dart [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+final Map<VeggieCategory, String> veggieCategoryNames = {
+  VeggieCategory.legume: 'Legume',
+  VeggieCategory.tuber: 'Tuber',
+  VeggieCategory.cruciferous: 'Cruciferous',
+  VeggieCategory.gourd: 'Gourd',
+  VeggieCategory.leafy: 'Leafy',
+  VeggieCategory.berry: 'Berry',
+  VeggieCategory.stealthFruit: 'Stealth fruit',
```
Comment (efortuna) [inline veggieseasons/lib/data/veggie.dart]: 😂😂😂

File: veggieseasons/lib/main.dart [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    DeviceOrientation.portraitDown,
+  ]);
+  runApp(new MyApp());
+}
+
+class MyApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return CupertinoApp(
+      color: Color(0xffd0d0d0),
```
Comment (efortuna) [inline veggieseasons/lib/main.dart]: this is due to my personal preference of disliking random hex numbers, but consider defining a variable that at least describes what color this is. Or add an in-line comment to that effect

File: veggieseasons/lib/screens/details.dart [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import 'package:flutter/widgets.dart';
+
+class DetailsScreen extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return CupertinoPageScaffold(
+      navigationBar: CupertinoNavigationBar(
+        middle: Text('Details'),
+      ),
+      backgroundColor: Color(0xfff0f0f0),
```
Comment (efortuna) [inline veggieseasons/lib/screens/details.dart]: same on the hex value here. Maybe put all these in an "iOSColors.dart" file? or add to that styles.dart file.

File: veggieseasons/lib/screens/favorites.dart [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import 'package:flutter/widgets.dart';
+
+class FavoritesScreen extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return CupertinoPageScaffold(
+      navigationBar: CupertinoNavigationBar(
+        middle: Text('My Garden'),
+      ),
+      backgroundColor: Color(0xfff0f0f0),
```
Comment (efortuna) [inline veggieseasons/lib/screens/favorites.dart]: ditto

File: veggieseasons/lib/screens/home.dart [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+          icon: Icon(CupertinoIcons.search),
+          title: Text('Search'),
+        ),
+        BottomNavigationBarItem(
+          icon: Icon(CupertinoIcons.settings),
+          title: Text('Settings'),
+        ),
+      ]),
+      tabBuilder: (context, index) {
+        if (index == 0) {
```
Comment (efortuna) [inline veggieseasons/lib/screens/home.dart]: nit: switch statement?

File: veggieseasons/lib/screens/list.dart [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+import 'package:flutter/widgets.dart';
+
+class ListScreen extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return CupertinoPageScaffold(
+      navigationBar: CupertinoNavigationBar(
+        middle: Text('List'),
+      ),
+      backgroundColor: Color(0xfff0f0f0),
```
Comment (efortuna) [inline veggieseasons/lib/screens/list.dart]: and here.

File: veggieseasons/lib/screens/search.dart [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+class SearchScreen extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return CupertinoPageScaffold(
+      navigationBar: CupertinoNavigationBar(
+        middle: Text('Search'),
+      ),
+      backgroundColor: Color(0xfff0f0f0),
+      child: Center(
+        child: Text('Not yet implemented.'),
```
Comment (efortuna) [inline veggieseasons/lib/screens/search.dart]: personal preference alert -- I'm not sure how big you expect these different files to get, but having a separate file for *every* class feels very Java-y. But you know best how these are going to grow. Just something to think about.

File: veggieseasons/lib/data/veggie.dart (hunk 1 above)
Comment (RedBrogdon) [reply veggieseasons/lib/data/veggie.dart]: There's a lot of them, as it turns out!

File: veggieseasons/lib/main.dart (hunk 2 above)
Comment (RedBrogdon) [reply veggieseasons/lib/main.dart]: Nice catch. Fixed.

File: veggieseasons/lib/screens/details.dart (hunk 3 above)
Comment (RedBrogdon) [reply veggieseasons/lib/screens/details.dart]: I've got a styles.dart class going that I hope to replace with something more sophisticated in a future release. I'll put it in there for now.

File: veggieseasons/lib/screens/favorites.dart (hunk 4 above)
Comment (RedBrogdon) [reply veggieseasons/lib/screens/favorites.dart]: Fixed.

File: veggieseasons/lib/screens/home.dart (hunk 5 above)
Comment (RedBrogdon) [reply veggieseasons/lib/screens/home.dart]: I actually had a switch statement in there to start, but took it out after seeing this:

https://www.dartlang.org/guides/language/language-tour#switch-and-case

Your Dart expertise trumps mine, though. Should I change it back?

File: veggieseasons/lib/screens/list.dart (hunk 6 above)
Comment (RedBrogdon) [reply veggieseasons/lib/screens/list.dart]: Fixed.

File: veggieseasons/lib/screens/search.dart (hunk 7 above)
Comment (RedBrogdon) [reply veggieseasons/lib/screens/search.dart]: You can see the full versions in https://github.com/redbrogdon/veggieseasons, if you want. They're not huge, but big enough that I figured separating them was the way to go.

**Pull request merged**